templates/storageclass-kadalu.yaml.j2

```diff
@@ -5,3 +5,4 @@ apiVersion: storage.k8s.io/v1
 metadata:
   name: kadalu
 provisioner: kadalu
+allowVolumeExpansion: true
```

Comment:
Should we add this option in 'external' storage storage class too? (Technically, we are already supporting it).

I am OK to do that as a separate patch, as we do it with CLI and also in documentation, but not in the code anywhere.

Reply:
### TEST

Created sample-pvc with 100Mi as claim with allowVolumeExpansion: true in stroage class
```
vatsa@vatsa-VirtualBox:~/kadalu$ sudo kubectl describe pvc pv1
Name:          pv1
Namespace:     default
StorageClass:  kadalu.replica1
Status:        Bound
Volume:        pvc-10cbbc2a-8591-4db6-a6ff-11326ed4cd2a
Labels:        <none>
Annotations:   kubectl.kubernetes.io/last-applied-configuration:
                 {"apiVersion":"v1","kind":"PersistentVolumeClaim","metadata":{"annotations":{},"name":"pv1","namespace":"default"},"spec":{"accessModes":[...
               pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: kadalu
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      100Mi
Access Modes:  RWO
VolumeMode:    Filesystem
Mounted By:    <none>
Events:
  Type    Reason                 Age   From                                                                  Message
  ----    ------                 ----  ----                                                                  -------
  Normal  ExternalProvisioning   11s   persistentvolume-controller                                           waiting for a volume to be created, either by external provisioner "kadalu" or manually created by system administrator
  Normal  Provisioning           11s   kadalu_kadalu-csi-provisioner-0_cd9bf8b4-7f11-42f3-9a11-57afae19625b  External provisioner is provisioning volume for claim "default/pv1"
  Normal  ProvisioningSucceeded  8s    kadalu_kadalu-csi-provisioner-0_cd9bf8b4-7f11-42f3-9a11-57afae19625b  Successfully provisioned volume pvc-10cbbc2a-8591-4db6-a6ff-11326ed4cd2a
```
Then patched for higher claim of 200Mi

```
vatsa@vatsa-VirtualBox:~/kadalu$ sudo kubectl patch pvc pv1 -p '{"spec":{"resources":{"requests":{"storage":"200Mi"}}}}}'
persistentvolumeclaim/pv1 patched
```
Logs from describe, csi-resizer and quotad
```
vatsa@vatsa-VirtualBox:~/kadalu$ sudo kubectl describe pvc pv1
Name:          pv1
Namespace:     default
StorageClass:  kadalu.replica1
Status:        Bound
Volume:        pvc-10cbbc2a-8591-4db6-a6ff-11326ed4cd2a
Labels:        <none>
Annotations:   kubectl.kubernetes.io/last-applied-configuration:
                 {"apiVersion":"v1","kind":"PersistentVolumeClaim","metadata":{"annotations":{},"name":"pv1","namespace":"default"},"spec":{"accessModes":[...
               pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: kadalu
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      200Mi
Access Modes:  RWO
VolumeMode:    Filesystem
Mounted By:    pod1
Events:
  Type     Reason                  Age    From                                                                  Message
  ----     ------                  ----   ----                                                                  -------
  Normal   ExternalProvisioning    4m35s  persistentvolume-controller                                           waiting for a volume to be created, either by external provisioner "kadalu" or manually created by system administrator
  Normal   Provisioning            4m35s  kadalu_kadalu-csi-provisioner-0_cd9bf8b4-7f11-42f3-9a11-57afae19625b  External provisioner is provisioning volume for claim "default/pv1"
  Normal   ProvisioningSucceeded   4m32s  kadalu_kadalu-csi-provisioner-0_cd9bf8b4-7f11-42f3-9a11-57afae19625b  Successfully provisioned volume pvc-10cbbc2a-8591-4db6-a6ff-11326ed4cd2a
  Warning  ExternalExpanding       80s    volume_expand                                                         Ignoring the PVC: didn't find a plugin capable of expanding the volume; waiting for an external controller to process this PVC.
  Normal   Resizing                80s    external-resizer kadalu                                               External resizer is resizing volume pvc-10cbbc2a-8591-4db6-a6ff-11326ed4cd2a
  Normal   VolumeResizeSuccessful  77s    external-resizer kadalu                                               Resize volume succeeded
```
```
vatsa@vatsa-VirtualBox:~/kadalu$ sudo kubectl logs -nkadalu kadalu-csi-provisioner-0 csi-resizer
I0114 14:45:03.136339       1 main.go:79] Version : v1.0.0-0-gb22717d-dirty
I0114 14:45:03.225380       1 connection.go:153] Connecting to unix:///var/lib/csi/sockets/pluginproxy/csi.sock
I0114 14:45:03.275644       1 common.go:111] Probing CSI driver for readiness
W0114 14:45:03.278921       1 metrics.go:142] metrics endpoint will not be started because `metrics-address` was not specified.
I0114 14:45:03.291063       1 controller.go:117] Register Pod informer for resizer kadalu
I0114 14:45:03.302767       1 controller.go:241] Starting external resizer kadalu
I0114 14:53:34.598992       1 event.go:282] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"pv1", UID:"10cbbc2a-8591-4db6-a6ff-11326ed4cd2a", APIVersion:"v1", ResourceVersion:"2865496", FieldPath:""}): type: 'Normal' reason: 'Resizing' External resizer is resizing volume pvc-10cbbc2a-8591-4db6-a6ff-11326ed4cd2a
I0114 14:53:37.506997       1 event.go:282] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"pv1", UID:"10cbbc2a-8591-4db6-a6ff-11326ed4cd2a", APIVersion:"v1", ResourceVersion:"2865496", FieldPath:""}): type: 'Normal' reason: 'VolumeResizeSuccessful' Resize volume succeeded
```
```
vatsa@vatsa-VirtualBox:~/kadalu$ sudo kubectl logs -nkadalu server-storage-pool-1-0-0 quotad
+ pid=0
+ cmd=/usr/bin/python3
+ script=/kadalu/server.py
+ trap 'kill ${!}; term_handler' SIGTERM
+ pid=6
+ true
+ wait 7
+ tail -f /dev/null
+ /usr/bin/python3 /kadalu/server.py
[2021-01-14 14:50:22,656] INFO [quotad - 105:handle_quota] - Quota set for size	 size=104857600
[2021-01-14 14:53:37,227] INFO [quotad - 105:handle_quota] - Quota set for size	 size=209715200
```

Reply:
I have not tested with an external-configuration yet so I don't know.

Reply:
should I remove this field for external-storage class?

Reply:
Let it be there for now!